cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp

```diff
@@ -0,0 +1,730 @@
+                            std::int64_t column_count,
+                            std::int64_t column_block_idx,
+                            std::int64_t column_block_count,
+                            std::int64_t tid,
+                            std::int64_t tnum) {
+    const std::int64_t col_offset = column_block_idx * tnum;
+    const std::int64_t x = tid + col_offset;
+
+    if (x < column_count) {
+        std::int64_t row_block_size = (row_count + row_block_count - 1) / row_block_count;
+        const std::int64_t row_offset = row_block_size * row_block_idx;
+
+        if (row_block_size + row_offset > row_count) {
+            row_block_size = row_count - row_offset;
+        }
+
+        Float sum = Float(0);
+        Float mean = Float(0);
+        Float sum2cent = Float(0);
+
+        for (std::int64_t row = 0; row < row_block_size; ++row) {
+            const std::int64_t y = (row + row_offset) * column_count;
+            const Float el = data_ptr[y + x];
+            Float inv_n = Float(1) / (row + 1);
+            Float delta = el - mean;
+
+            sum += el;
+            mean += delta * inv_n;
+            sum2cent += delta * (el - mean);
+        }
+
+        bsum_ptr[x * row_block_count + row_block_idx] = sum;
+
+        bsum2cent_ptr[x * row_block_count + row_block_idx] = sum2cent;
+    }
+}
+
+/* block processing kernel for device execution */
+template <typename Float, bool DefferedFin>
+inline void merge_blocks_kernel(sycl::nd_item<1> item,
+                                const std::int64_t* brc_ptr,
+                                const Float* bsum_ptr,
+                                const Float* bsum2cent_ptr,
+                                std::int64_t* lrc_ptr,
+                                Float* lsum_ptr,
+                                Float* lsum2cent_ptr,
+                                Float* lmean_ptr,
+                                Float* rsum_ptr,
+                                Float* rsum2cent_ptr,
+                                Float* rmean_ptr,
+                                Float* rvarc_ptr,
+                                std::int64_t id,
+                                std::int64_t group_id,
+                                std::int64_t local_size,
+                                std::int64_t block_count) {
+    Float mrgsum = Float(0);
+    Float mrgmean = Float(0);
+    Float mrgvectors = Float(0);
+    Float mrgsum2cent = Float(0);
+    lrc_ptr[id] = 0;
+
+    for (std::int64_t i = id; i < block_count; i += local_size) {
+        std::int64_t offset = group_id * block_count + i;
+
+        Float sum = Float(0);
+        sum = bsum_ptr[offset];
+
+        std::int64_t rcnt = 1;
+        rcnt = brc_ptr[offset];
+
+        Float sum2cent = Float(0);
+        sum2cent = bsum2cent_ptr[offset];
+
+        Float mean = sum / static_cast<Float>(rcnt);
+
+        const Float sum_n1n2 = mrgvectors + static_cast<Float>(rcnt);
```

Comment:
I see inconsistency between using `Float(1)` and `static_cast<Float>(rcnt)`... Please use the only way to convert data types

Reply:
Btw you can use `Float{}` instead of `Float(0)`.